### PR TITLE
Added fixes which caused issues building on Apple Silicon

### PR DESCRIPTION
**Pull request by mcbarton**
Stopped Cuda library been loaded when on Apple
Add .dylib option for when on MacOS

### CODE REVIEW

File: cmake/CppInterOp/CppInterOpConfig.cmake.in [hunk 1]
```diff
@@ -6,6 +6,13 @@ get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}"
 get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)
 get_filename_component(CPPINTEROP_INSTALL_PREFIX "${CPPINTEROP_INSTALL_PREFIX}" PATH)
 
+# Determine CMAKE_SHARED_LIBRARY_SUFFIX based on operating system
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  set(CMAKE_SHARED_LIBRARY_SUFFIX  ".dylib")
```
Comment (vgvassilev) [inline cmake/CppInterOp/CppInterOpConfig.cmake.in:11]: Are the changes introduced in Is https://github.com/compiler-research/CppInterOp/commit/4eac1caa49d86ce9e51da1d46bfe38675f57444c not sufficient?

Comment (mcbarton) [reply cmake/CppInterOp/CppInterOpConfig.cmake.in:11]: I didn't find it to be sufficient on my machine.

Comment (vgvassilev) [reply cmake/CppInterOp/CppInterOpConfig.cmake.in:11]: cc: @alexander-penev

Comment (vgvassilev) [reply cmake/CppInterOp/CppInterOpConfig.cmake.in:11]: @mcbarton, there is some discussion about `.so` vs `.dylib`: https://gitlab.kitware.com/cmake/cmake/-/issues/21189. They say that `macOS distinguishes between shared libraries and loadable modules.  CMAKE_SHARED_LIBRARY_SUFFIX is set to .dylib for shared libraries.  CMAKE_SHARED_MODULE_SUFFIX is set to .so for loadable modules (e.g. dlopen).  See the [add_library](https://cmake.org/cmake/help/v3.18/command/add_library.html#normal-libraries) command's SHARED and MODULE options.`.

CppInterOp is meant to be dlopened thus it means that we should keep probably the `.so` extension. We can do that by adding the `MODULE` option to https://github.com/compiler-research/CppInterOp/blob/4eac1caa49d86ce9e51da1d46bfe38675f57444c/lib/Interpreter/CMakeLists.txt#L87C1-L93

Comment (mcbarton) [reply cmake/CppInterOp/CppInterOpConfig.cmake.in:11]: @vgvassilev I tried your suggestion and it didn't work on my machine. Adding the MODULE option caused the following to occur
1) The following error is produced during cmake configure stage
CMake Error at unittests/CppInterOp/CMakeLists.txt:16 (target_link_libraries):
  Target "clangCppInterOp" of type MODULE_LIBRARY may not be linked into
  another target.  One may link only to INTERFACE, OBJECT, STATIC or SHARED
  libraries, or to executables with the ENABLE_EXPORTS property set.
 
 
CMake Error at unittests/CppInterOp/CMakeLists.txt:26 (target_link_libraries):
  Target "clangCppInterOp" of type MODULE_LIBRARY may not be linked into
  another target.  One may link only to INTERFACE, OBJECT, STATIC or SHARED
  libraries, or to executables with the ENABLE_EXPORTS property set.
2) dylib extension stayed (determined by turning off building of unittests)
3) libclangCppInterop.dylib got renamed to clangCppInterop.dylib for some reason (determined by turning off building of unittests)

Any other suggestions?

Comment (vgvassilev) [reply cmake/CppInterOp/CppInterOpConfig.cmake.in:11]: Ok, then we are back to your original patch. The part I do not understand is why then the `CMAKE_SHARED_LIBRARY_SUFFIX` variable is not set. I'd rather have `include(CMakeSystemSpecificInformation)` than setting it by hand. I checked on my platform and that does set the right suffix. Can we try that?

**Review by vgvassilev — COMMENTED** [review]
LGTM, can you run git-clang-format to the diff? That is, squash the commits into one and then run `git-clang-format HEAD~`.

**Comment by mcbarton** [comment]
@vgvassilev I have squashed both commits into one, and then performed the command git-clang-format HEAD~, and committed this. I think you meant me to do this in such a way as to reduce everything to a single commit, but was unsure. Will squash the 2 commits including the git-clang-format one into 1 once you confirm?

**Comment by vgvassilev** [comment]
> @vgvassilev I have squashed both commits into one, and then performed the command git-clang-format HEAD~, and committed this. I think you meant me to do this in such a way as to reduce everything to a single commit, but was unsure. Will squash both commits of this pull request into 1 once you confirm.

Now I can do it on my end.

**Pull request merged**